ip_usage.go

```diff
@@ -58,10 +57,15 @@ func updateIPUsage(r *http.Request) error {
 	if err != nil {
 		return err
 	}
-	dbResult := db.Where(&ipUse{IP: ip}).Assign(&ipUse{LastUse: time.Now()}).FirstOrCreate(&ipUse{})
-	dbErrors := dbResult.GetErrors()
-	if httpserverutils.HasDBError(dbErrors) {
-		return httpserverutils.NewErrorFromDBErrors("Some errors were encountered when upserting the IP to the new date:", dbResult.GetErrors())
+
+	ipUse := &ipUse{IP: ip}
+	_, err = db.Model(ipUse).
+		OnConflict("(ip) DO UPDATE").
+		Set("last_use = ?", time.Now()).
```

Comment:
Just making sure: the Set will be triggered both on conlfict and not on conflict, right?
Maybe we want to set last_use inside `ipUse` to start with, to be more explicit either way?

Reply:
Yes, it will set it either way. But your suggestion is more elegant, fixed.